src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -65,7 +59,7 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
     	 DmlWrapper dmlWrapper = new DmlWrapper();
 		 List<Affiliation__c> afflsMadePrimary = new List<Affiliation__c>();
 
-		 if(!afflMultiHasRunBefore) {
+		 if(!TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_Before)) {
```

Comment:
The naming of these flags is misleading since the enclosed code handles Before Insert and After Update. Also note that ALL configured events will get past this IF statement the first time since it doesn't actually check the triggerAction until line 92. That also means line 133 is called for ALL events because there's no triggerAction check around that line before setting that flag to true. Ideally this IF statement would also check to see if this is a Before Insert or After Update